### PR TITLE
Add pre-commit hook

### PR DESCRIPTION
**Pull request by cassava**
I recommend reviewing this commit-by-commit, instead of all at once.

### CODE REVIEW

**Review by tobifalk — APPROVED** [review]
LGTM!

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        files: ^cli/
+  - repo: https://github.com/psf/black
+    rev: 20.8b1
+    hooks:
+      - id: black
+        args:
+          - --quiet
+          - --safe
+        types: [python]
+        files: ^cli/
```
[the remark this answers is not in the document]
Comment (cassava) [reply .pre-commit-config.yaml:62]: The linters complained about a lot of things that didn't really make a lot of sense for Conanfiles, and I didn't want to force that on the devs yet.

[the remark this answers is not in the document]
Comment (cassava) [reply .pre-commit-config.yaml:62]: We can always expand the domain later.

File: ui/src/components/chart.jsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ import { Scatter } from "react-chartjs-2";
 import "chartjs-plugin-zoom";
 
 // The Chart component creates a chart out of data.
-// Everytime the props 'nextXValue' and 'nextYValues' are changing,
+// Every time the props 'nextXValue' and 'nextYValues' are changing,
 // the component adds them to it"s data set and renders the new state.
```
Comment (clssn) [inline ui/src/components/chart.jsx:22]: ... its data set ...

**Review by clssn — APPROVED** [review]
Got a single minor finding slightly out of scope. You can fix it now or note it down for later resolution.
Your changes look good and I'm especially fond of codespell to find so many mistakes! :+1:

File: ui/src/components/chart.jsx (hunk 2 above)
Comment (cassava) [reply ui/src/components/chart.jsx:22]: I'd rather do this later.

**Pull request merged**